platform/api/aws/ec2.go

```diff
@@ -255,12 +283,192 @@ func (a *API) getSecurityGroupID(name string) (string, error) {
 	return *sgIds.SecurityGroups[0].GroupId, nil
 }
 
+// getVPCID gets a VPC for the given security group
+func (a *API) getVPCID(sgId string) (string, error) {
+	sgs, err := a.ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: []*string{&sgId},
+	})
+	for _, sg := range sgs.SecurityGroups {
+		if sg.VpcId != nil {
+			return *sg.VpcId, nil
+		}
+	}
+	return "", fmt.Errorf("no vpc found for security group %v: %v", sgId, err)
+}
+
+// getSubnetID gets a subnet for the given VPC.
+func (a *API) getSubnetID(vpc string) (string, error) {
+	subIds, err := a.ec2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{&vpc},
+			},
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("unable to get subnets for vpc %v: %v", vpc, err)
+	}
+	for _, id := range subIds.Subnets {
+		if id.SubnetId != nil {
+			return *id.SubnetId, nil
+		}
+	}
+	return "", fmt.Errorf("no subnets found for vpc %v: %v", vpc, err)
+}
+
+// creates an InternetGateway and attaches it to the given VPC
+func (a *API) createInternetGateway(vpcId *string) (string, error) {
+	igw, err := a.ec2.CreateInternetGateway(&ec2.CreateInternetGatewayInput{})
+	if err != nil {
+		return "", err
+	}
+	a.CreateTags([]string{*igw.InternetGateway.InternetGatewayId}, map[string]string{
```

Comment:
nit: I notice the ``"CreatedBy": "mantle"`` is used multiple times. In the future it may be worth making a tags variable which gets reused for these calls. No biggie though. Just a thought.